test/e2e/app_namespace_test.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2023 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package e2e
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestAppNamespace(t *testing.T) {
+	env := BuildEnv(t)
+	logger := Logger{}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
+	kubectl := Kubectl{t, env.Namespace, logger}
+
+	name := "test-app-namespace"
+
+	cleanUp := func() {
+		kapp.Run([]string{"delete", "-a", name})
+	}
+
+	cleanUp()
+	defer cleanUp()
+
+	yaml := `
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: test-cm
+`
+
+	logger.Section("deploy app with -n flag", func() {
+		kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name, "-n", env.Namespace}, RunOpts{NoNamespace: true, StdinReader: strings.NewReader(yaml)})
+
+		// both app meta configmap and the resources should be present in the -n namespace
+		NewPresentClusterResource("configmap", name, env.Namespace, kubectl)
+		NewPresentClusterResource("configmap", "test-cm", env.Namespace, kubectl)
+	})
+
+	logger.Section("deploy same app with both -n and --app-namespace flag", func() {
```

Comment:
I imagine we might have a case where we want to put resources with no ns specified to one location and have the app CM present in one.

Reply:
Are we saying that migration of app CM is not allowed? Could use some clarity here :/

Reply:
> I imagine we might have a case where we want to put resources with no ns specified to one location and have the app CM present in one.

That would be testing the same thing, right? I mean I can create a service account and set the preferred namespace to env.Namespace, but that would be same as using -n env.Namespace.

Reply:
> Are we saying that migration of app CM is not allowed?

What kind of migration are we talking about? Like creating an app in namespace A and then creating in namespace B? (That would be a different app)